src/recommendations/data/tests/hooks.test.jsx

```diff
@@ -0,0 +1,73 @@
+import { renderHook } from '@testing-library/react-hooks';
+import algoliasearchHelper from 'algoliasearch-helper';
+
+import mockedAlgoliaCourses from './mockedData';
+import CreateAlgoliaSearchHelperMock from './test_utils/test_utils';
+import isOneTrustFunctionalCookieEnabled from '../../../data/utils/oneTrust';
+import useAlgoliaRecommendations from '../hooks/useAlgoliaRecommendations';
+import useStaticRecommendations from '../hooks/useStaticRecommendations';
+
+jest.mock('algoliasearch-helper');
+
+jest.mock('../../../data/utils/oneTrust');
+
+jest.mock('../../../data/algolia', () => ({
+  initializeSearchClient: jest.fn(),
+  getPrecisionRadius: jest.fn(() => 1609 * 50),
+  getLocationRestrictionFilter: jest.fn((countryCode) => `NOT BLOCKED IN ${countryCode}`),
+}));
+
+jest.mock('../algoliaResultsParser', () => jest.fn((course) => course));
+
+describe('HooksTests', () => {
+  describe('useAlgoliaRecommendations Tests', () => {
+    const MockSearchHelperWithData = new CreateAlgoliaSearchHelperMock(mockedAlgoliaCourses);
+    const MockSearchHelperWithoutData = new CreateAlgoliaSearchHelperMock();
+
+    it('should fetch recommendations only if functional cookies are set', async () => {
+      isOneTrustFunctionalCookieEnabled.mockImplementation(() => true);
+      algoliasearchHelper.mockImplementation(() => MockSearchHelperWithData);
+      const { result } = renderHook(
+        () => useAlgoliaRecommendations('PK', 'Introductory'),
+      );
+
+      expect(result.current.recommendations).toEqual(mockedAlgoliaCourses);
+      expect(result.current.isLoading).toBe(false);
+    });
+
+    it('should not fetch recommendations if functional cookies are not set', async () => {
+      isOneTrustFunctionalCookieEnabled.mockImplementation(() => false);
+      algoliasearchHelper.mockImplementation(() => MockSearchHelperWithData);
+      const { result } = renderHook(
+        () => useAlgoliaRecommendations('PK', 'Introductory'),
+      );
+
+      expect(result.current.recommendations).toEqual([]);
+      expect(result.current.isLoading).toBe(false);
+    });
+
+    it('should return empty list if no recommendations returned from Algolia', async () => {
+      isOneTrustFunctionalCookieEnabled.mockImplementation(() => true);
+      algoliasearchHelper.mockImplementation(() => MockSearchHelperWithoutData);
+      const { result } = renderHook(
+        () => useAlgoliaRecommendations('PK', 'Introductory'),
+      );
+
+      expect(result.current.recommendations).toEqual([]);
+      expect(result.current.isLoading).toBe(false);
+    });
+  });
+
+  describe('useStaticRecommendations Tests', () => {
+    it('should return trending and popular products list', async () => {
+      jest.spyOn(JSON, 'parse').mockReturnValue(mockedAlgoliaCourses);
```

Comment:
mocked data object `mockedAlgoliaCourses`  name should be generic or should have a separate object for static recommendations.

Reply:
yes you are right. i will update the name.